### PR TITLE
Support geometry collection in Quadint polyfill

### PR DESCRIPTION
**Pull request by vdelacruzb**

### CODE REVIEW

File: modules/quadkey/bigquery/sql/ST_ASQUADINT_POLYFILL.sql [hunk 1]
```diff
@@ -17,6 +17,22 @@ AS """
...
+CREATE OR REPLACE FUNCTION `@@BQ_PREFIX@@quadkey.__geometrycollection`
+(geojson STRING)
+RETURNS STRING
+DETERMINISTIC
+LANGUAGE js
+OPTIONS (library=["@@BQ_LIBRARY_BUCKET@@"])
+AS """
+    const pol = JSON.parse(geojson);
+    return JSON.stringify(quadkeyLib.geometrycollection(pol))
+    // Iterate through collection features
```
Comment (Jesus89) [inline modules/quadkey/bigquery/sql/ST_ASQUADINT_POLYFILL.sql:29]: Why are all the comments here?

Comment (vdelacruzb) [reply modules/quadkey/bigquery/sql/ST_ASQUADINT_POLYFILL.sql]: It was early WIP, just in case Alex had to continue. The function works now, just have to add integration tests.

By the way @alex-graciano I have seen that we have the same problem with H3 polyfill. Do you want to tackle that one? It's not crashing but it returns an empty array.

**Comment by vdelacruzb** [comment]
Made tests in SF use `ST_ASQUADINT_POLYFILL` and also added duplicated features to tests to double check that the resulting array doesn't change as we remove duplicates.

**Review by Jesus89 — APPROVED** [review]
LGTM!

**Comment by Jesus89** [comment]
I've tested this in staging and found the following error:

```sql
SELECT quadkey.ST_ASQUADINT_POLYFILL(geom,16) FROM `cartobq.testtables.polygons_800k` LIMIT 1000
-- Error: NULL argument passed to UDF at UDF$1(STRING, INT64) line 3, columns 8-9
```

This is because the dataset contains `GEOMETRYCOLLECTION EMPTY` and this ends up in NULL GeoJSON. I think we should cover this case to avoid the bug, and return `[]` in that case. cc @martejpad

**Comment by vdelacruzb** [comment]
> I've tested this in staging and found the following error:
> 
> ```sql
> SELECT quadkey.ST_ASQUADINT_POLYFILL(geom,16) FROM `cartobq.testtables.polygons_800k` LIMIT 1000
> -- Error: NULL argument passed to UDF at UDF$1(STRING, INT64) line 3, columns 8-9
> ```
> 
> This is because the dataset contains `NULL` geographies and this ends up in `NULL` GeoJSON. I think we should cover this case to avoid the bug, and return `[]` in that case. cc @martejpad

Definitely I remember that in some cases we are returning errors in Quadint but for this one, it should be an empty array. Updating it.

**Comment by vdelacruzb** [comment]
The NULL issue is related with https://github.com/CartoDB/carto-spatial-extension/issues/148.

I thought Bigquery's [Safe Prefix](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#safe_prefix) was also working with UDFs. Should we request that? If it was supported all the functions should return Error by default.

**Comment by Jesus89** [comment]
OK then. We can merge it as it is and create a ticket in CH to update this behavior in all the functions if required and possible.

**Comment by Jesus89** [comment]
But remember to update the changelog first
